src/overlay/test/OverlayTests.cpp

```diff
@@ -1734,4 +1736,66 @@ TEST_CASE("peer is purged from database after few failures",
 
     REQUIRE(!peerManager.load(localhost(cfg2.PEER_PORT)).second);
 }
+
+TEST_CASE("generalized tx sets are not sent to non-upgraded peers",
+          "[txset][overlay]")
+{
+    if (protocolVersionIsBefore(Config::CURRENT_LEDGER_PROTOCOL_VERSION,
+                                GENERALIZED_TX_SET_PROTOCOL_VERSION))
+    {
+        return;
+    }
+    auto runTest = [](bool hasNonUpgraded) {
+        int const nonUpgradedNodeIndex = 1;
+        auto networkID = sha256(getTestConfig().NETWORK_PASSPHRASE);
+        auto simulation = Topologies::core(
+            4, 0.75, Simulation::OVER_LOOPBACK, networkID, [&](int i) {
+                auto cfg = getTestConfig(i, Config::TESTDB_ON_DISK_SQLITE);
+                cfg.MAX_SLOTS_TO_REMEMBER = 10;
+                cfg.TESTING_UPGRADE_LEDGER_PROTOCOL_VERSION =
+                    static_cast<uint32_t>(GENERALIZED_TX_SET_PROTOCOL_VERSION);
+                if (hasNonUpgraded && i == nonUpgradedNodeIndex)
+                {
+                    cfg.OVERLAY_PROTOCOL_VERSION =
+                        Peer::FIRST_VERSION_SUPPORTING_GENERALIZED_TX_SET - 1;
+                }
+                return cfg;
+            });
+
+        simulation->startAllNodes();
+        auto nodeIDs = simulation->getNodeIDs();
+        auto node = simulation->getNode(nodeIDs[0]);
+
+        auto root = TestAccount::createRoot(*node);
+
+        int64_t const minBalance =
```

Comment:
similarly here, you can just use loadgen to generate transactions

Reply:
As per additional offline discussion, removed this test and the logic for handling non-upgraded peers. Non-upgraded peers won't be able to parse new XDR anyway, so we don't need to specially handle them.